src/java/org/apache/cassandra/service/accord/AccordCache.java

```diff
@@ -1131,13 +1136,30 @@ public Object fullShrink(RoutingKey key, CommandsForKey value)
             if (value.isLoadingPruned())
                 return value;
 
-            return Serialize.toBytesWithoutKey(value.maximalPrune());
+            TxnId last = value.size() == 0 ? null : value.get(value.size() - 1);
+            TxnId minUndecided = value.minUndecided();
+            int prefixBytes = (int) (  CommandSerializers.txnId.serializedSize(last)
+                                     + CommandSerializers.txnId.serializedSize(minUndecided));
+            ByteBuffer result = Serialize.toBytesWithoutKey(prefixBytes, value.maximalPrune());
+            CommandSerializers.txnId.serialize(last, result);
+            CommandSerializers.txnId.serialize(minUndecided, result);
+            result.position(0);
+            return result;
+        }
+
+        private static int prefixBytes(ByteBuffer bb)
+        {
+            int prefix = (int) CommandSerializers.txnId.serializedSize(CommandSerializers.txnId.deserialize(bb, 0));
+            prefix += (int)CommandSerializers.txnId.serializedSize(CommandSerializers.txnId.deserialize(bb, prefix));
+            return prefix;
         }
 
         @Override
         public CommandsForKey inflate(AccordCommandStore commandStore, RoutingKey key, Object shrunk)
         {
-            return Serialize.fromBytes(key, (ByteBuffer)shrunk);
+            ByteBuffer bb = (ByteBuffer)shrunk;
+            bb.position(prefixBytes(bb));
```

Comment:
Should we duplicate here maybe?

Reply:
yep, good idea. we should be discarding it, but better safe than sorry and can do it without any net cost by skipping the internal duplicate